ui/src/components/ai-chat/component/operation-button/LogOperationButton.vue

```diff
@@ -177,7 +178,12 @@ const playAnswerText = (text: string) => {
     }
     applicationApi
       .postTextToSpeech(id || (props.applicationId as string), { text: text }, loading)
-      .then((res: any) => {
+      .then(async (res: any) => {
+        if (res.type === 'application/json') {
+          const text = await res.text()
+          MsgError(text)
+          return
+        }
         // 假设我们有一个 MP3 文件的字节数组
         // 创建 Blob 对象
         const blob = new Blob([res], { type: 'audio/mp3' })
```

Comment:
There is one main issue in the provided code snippet:

1. **Potential Memory Leak with `MsgError`:** If a network request fails or returns an unexpected response that still contains HTML/XML content, the `MsgError` function might attempt to parse it, leading to unnecessary memory allocations.

2. **Optimization:** Since you are using `await res.text()` when the response type is JSON, consider checking the contentType header of the response before attempting to read body content as text. This can prevent unnecessary conversions and improve performance.

3. **Code Structure Review:**
   ```typescript
   const playAnswerText = (text: string) => {
     if (!response || !response.ok) {
       console.error('Network Error:', error);
       return;
     }
     
     switch (response.headers.get('Content-Type')) {
       case 'application/json':
         const jsonData = await response.json();
         displayJsonData(jsonData); // Your custom handling for JSON data
         break;
       default:
         downloadFileFromResponse(response);
         break;
     }
   }
 ```

### Suggested Changes:
- **Avoid Parsing Non-JSON Responses:** Before converting non-JSON responses to strings (`async res.text()`) check the response's Content-Type Header.
  
  **Updated Code Snippet:**

  ```typescript
  playAnswerText(text: string) {
    try {
      applicationApi.postTextToSpeech(
        id || props.applicationId,
        { text },
        false
      )
        .then(async (res) => {
          switch (res.status) {
            case 200:
              if (!res.headers.has('content-type') || /json$/.test(res.headers get('content-type'))) {
                const responseData = await res.json(); // Attempt to parse json even without specific MIME types.
                  ... handle json response ...
              } else {
                  ... process other content-types if needed...
              }
              // Continue normal execution after successful parsing
              ......
              
             // Downloading file from response if necessary.
             const url = window.URL.createObjectURL(new Blob([res]));
             const link = document.createElement("a");
             link.style.display = "none";
             link.href = url;
             link.download = "downloadedFile.mp3"; // Specify appropriate filename here.
             
             document.body.appendChild(link);
             link.click();

             window.URL.revokeObjectURL(url);

              break;

            case 4XX:
            case 5XX:
              const message = await res.text();
               MsgError(message);
              break;

            default:
              break; // Other status codes that you may need to handle appropriately.
          }
       
          
        })
        .catch(error => {
          console.error('Failed POST', error.message);
           MsgError(`Error ${error.message}`);
        });
      
        
    } catch (parseErr) {
      console.log(parseErr);
       MsgError(`Parse Error: ${parseErr}`);
  ```

### Summary:
The suggested changes address the memory leak risk by checking and dealing with non-json responses carefully, preventing unnecessary attempts at parsing them into text. Additionally, proper logging and clear separation between error states are maintained throughout the processing flow.